lib/lua/CMakeLists.txt

```diff
@@ -22,6 +22,10 @@ else()
 	set(DEFAULT_ANSI TRUE)
 endif()
 
+if(MSVC)
+	set(COMMON_CFLAGS "${COMMON_CFLAGS} -D_CRT_SECURE_NO_WARNINGS")
```

Comment:
Should this be `/D`? When you use add_definition like in IrrlichtMt CMake does the conversion itself but this just concatenates a raw flag into the flags variable like in the main Minetest codebase. FWIW test building without LuaJIT on MSVC to see if it works.

Reply:
I am using MT without LuaJIT currently, do you want me to change it anyways?
![image](https://github.com/user-attachments/assets/6caac92a-aff7-4650-a3e2-6c6632a96ccb)

It doesn't actually pass it as a command line flag to the linker but does the slightly smarter thing and adds it as definition in the project file.
![image](https://github.com/user-attachments/assets/cb328f9c-612d-4f4e-a09a-10408ad5a1a8)

Reply:
Either CMake does some conversion internally or it's inofficially supported by `CL`. `CL /HELP` might provide more information on whether it supports dash-prefixed options. Either way, `/D` is (as far as I can see) the correct notation for msvc.

https://learn.microsoft.com/en-us/cpp/build/reference/compiler-options-listed-alphabetically?view=msvc-170

Reply:
CMake handles it anyways since it needs to generate a project file - it's not invoking CL directly, basically no-one invokes CL directly since you might as well use Clang at that point.

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS.html

> These flags will be passed to all invocations of the compiler. This includes invocations that drive compiling and those that drive linking.

We do not invoke `CL` but msbuild does, with arguments originating from the project file.

> you might as well use Clang at that point.

I'm not sure which paths it would take in CMakeLists (such as the `if (MSVC)` check), but different compilers would generally need special treatment like already present in this file.

---

Based on your screenshot, it indeed confirms that such conversion happens *somewhere*. Either way, based on other occurrences of `-D` that are also used with MSVC, I suppose this change is OK.

Reply:
Project file does *not* store arguments (in this case) was my point since CMake is generating a project file for MSVC not calling `cl` using `make` as the build system or what have you.

It doesn't seem to be documented anywhere that CMake handles this correctly, but check the source code, they appear to detect flags that are actually definitions and handle them separately. Seems to be to support IDEs like MSVC.

https://github.com/Kitware/CMake/blob/0d0ae4b5772f3ea77bfc856eceaf87da82220fde/Source/cmMakefile.cxx#L1425

Should probably rewrite everything to use `add_definition` eventually anyways.